### PR TITLE
fix user avatar transformation - change number of buffs owned when you use them - fixes #9795

### LINKED ISSUE #9795: no change in number of avatar transformation buffs owned when you use them (Snowball, Shiny Seed, Seafoam, Spooky Sparkles)
When you use an avatar transformation buff items (e.g., Snowball) the number owned shown in your inventory doesn't decrease until you click the sync button.

It should decrease as soon as you use one item.

### PR DESCRIPTION
**Pull request by negue**
Cast the spell also on the website, to change the `stats.buffs` properties.

Fixes #9795

### CODE REVIEW

**Comment by TheHollidayInn** [comment]
Fixes #9795

File: website/client/mixins/spells.js [hunk 1]
```diff
@@ -73,12 +79,9 @@ export default {
...
       this.applyingAction = false;
 
-      let spellUrl = `/api/v3/user/class/cast/${spell.key}`;
-      if (targetId) spellUrl += `?targetId=${targetId}`;
-
       let spellText = typeof spell.text === 'function' ? spell.text() : spell.text;
 
-      await axios.post(spellUrl);
+      let apiResult = await this.$store.dispatch('user:castSpell', {key: spell.key, targetId});
       let msg = this.$t('youCast', {
```
Comment (paglias) [inline website/client/mixins/spells.js]: can we move this to a default branch in the switch just below?

**Review by paglias — APPROVED** [review]

**Review by TheHollidayInn — APPROVED** [review]

**Comment by paglias** [comment]
There's an issue with this @negue, when you buy a card and try to cast it on a party member the modal with the list of members appears but you can only cast if you click on the avatar, clicking on `Select` doesn't wok. Also the cursor becomes a cross which is not currently happening on staging

**Comment by negue** [comment]
huh odd, I think I only used the select button never the avatar itself, will check again

**Comment by negue** [comment]
@paglias  I couldn't reproduce the cast on avatar , it only oped the info modal , but I found a different issues which caused only to cast the spell every second try

**Comment by paglias** [comment]
@negue working now!

**Pull request merged**